openedx-unsupported/ecommerce
# feat: Mgmt Command to create mobile seats for new course runs

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)



## Anyone internally merging to this repository is expected to [release and monitor their changes](https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories); if you are not able to do this DO NOT MERGE, please coordinate with someone who can to ensure that the changes are released.

## Required Testing
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

This management command is to be configured as a cron job to be run on the ecommerce system. It fetches the products that have already been configured for payments on mobile. And then checks (from discovery) if new course runs have been created which do not have mobile SKUS added yet.
It then takes two actions:

1. Emails the mobile team course keys for these expired courses
2. Creates Seats and Mobile SKUS for these courses on Ecommerce and publishes these changes to LMS Course modes. 

## Supporting information

Jira link: https://2u-internal.atlassian.net/browse/LEARNER-9566

## Testing instructions

- Create a course, and course runs. 
- Feed this data in the discovery system. 
- Create mobile seats for a course but not its course run.
- Run the given management command on Ecommerce.
- Verify the mobile seats have been automatically created for the new Course run.
